Add an API to calculate payouts when the taker goes short

luckysori opened the pull request:
Fixes #1432.

luckysori committed:
Remove mention of nonexistent parameter in docstring

luckysori committed:
Allow configuration of leverage for both sides

We can move the constant that fixes the maker's leverage to an upper
layer. The `payout_curve` module doesn't need to have an opinion on
anyone's leverage, or knowledge of the concepts of maker and taker.

luckysori committed:
Use short/long over taker/maker in payout_curve.rs

luckysori committed:
Adapt docstring to reference all parameters

thomaseizinger commented on model/src/payout_curve.rs:
```diff
@@ -158,13 +182,21 @@ struct PayoutParameter {
         )
     }
+
+    fn into_payouts_short_taker(self) -> Result<Vec<Payout>> {
```
Why is this necessary? If the caller know the role of the position, wouldn't that be good enough?

luckysori replied:
It is not necessary but convenient. Before we had an `into_payouts` function which knew that the taker was always long. 

I can remove both functions and inline this logic.

thomaseizinger replied:
Shouldn't we change `maia` to care about `long` and `short` instead of `maker` and `taker`?

I am happy to merge this for now but we should fix this in maia I believe.

thomaseizinger replied:
I created a ticket for it: https://github.com/comit-network/maia/issues/16

luckysori replied:
> Shouldn't we change `maia` to care about `long` and `short` instead of `maker` and `taker`?

Ultimately yes.

da-kami commented on model/src/payout_curve.rs:
```diff
@@ -43,29 +41,63 @@ mod utils;
 ///
 /// The list of [`Payout`]s for the given price, quantity and leverage.
-pub fn calculate(
+pub fn calculate_long_taker(
```
Is the expectation that there will be a `calculate_short_taker` as well? 
Will there be separate functions for `calculate_long_maker` and `calculate_short_maker`? 

I would actually find it easier if we would move this logic on the `Cfd`. The cfd knows position and role and can decide how to call the `calculate_payout_parameters` parameters. I find the indirection with `calculate_long_taker` in between confusing.

luckysori replied:
Fair, I can try to move these APIs to the `model` crate.

klochowicz reviewed (approved):
I'm happy for this to go in, as it's an improvement. removing the "maker" and "taker" from this part of the code should probably be tackled along with `maia` refactor.

luckysori committed:
Define functions model::calculate_payouts_{long, short}_taker

At the moment, we only use `calculate_payouts_long_taker` in the
application because it is the only thing that is currently fully
supported.
